kordac/processors/ImageBlockProcessor.py

```diff
@@ -14,6 +15,8 @@ def __init__(self, ext, *args, **kwargs):
         self.template = ext.jinja_templates[self.processor]
         self.relative_image_template = ext.jinja_templates['relative-image-link']
         self.required = ext.required_files['images']
+        self.required_parameters = {'file_path'}
+        self.optional_parameters = {'alt': set(), 'caption': set(), 'caption-link': {'caption'}, 'source': set(), 'alignment': set(), 'hover-text':{}}
```

Comment:
Can you explain why `required_parameters` is a set? And why `optional_parameters` is a dictionary with values varying between an empty set, a set, and a dictionary?